### ISSUE TITLE
Request for clarification weight_layers function

### ISSUE THREAD

**Issue by charlesdognin** [allenai/bilm-tf#103]
Hi, 

Thanks a lot for the code. 
I am not sure though to understand how the weights and the gamma scale from the weight_layers function are trained. In particular they are linked to no optimizer. Does it mean that we have to run weight_layers inside the task specific training session so that those variables can be added to the list of trainable variables of our task specific model?

Thanks a lot in advance for your answer, 
Best,

**Comment by matt-peters** [comment]
Yes, that's the case.  This function is intended to be used as part of another model training loop (that defines an appropriate loss, etc).
